clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/UpsertGoogleAutoscalingPolicyAtomicOperation.groovy

```diff
@@ -180,18 +180,20 @@ class UpsertGoogleAutoscalingPolicyAtomicOperation extends GoogleAtomicOperation
     // TODO(jacobkiefer): Update metadata for autoHealingPolicy when 'mode' support lands.
     // NOTE: This block is here intentionally, we should wait until all the modifications are done before
     // updating the instance template metadata.
-    if (isRegional) {
-      updatePolicyMetadata(compute,
-                           credentials,
-                           project,
-                           GCEUtil.buildRegionalServerGroupUrl(project, region, serverGroupName),
-                           autoscaler)
-    } else {
-      updatePolicyMetadata(compute,
-                           credentials,
-                           project,
-                           GCEUtil.buildZonalServerGroupUrl(project, zone, serverGroupName),
-                           autoscaler)
+    if (description.writeMetadata == null || description.writeMetadata) {
```

Comment:
What's the actual path where this can end up `null`? Looks like the converter only overwrites the default `True` if a non-null value is passed in the request input.

Also, if it can in fact end up `null`, why does that mean we should update the metadata?

Reply:
This will be null if a user explicitly issues an upsert description with `writeMetadata = null`. The converter change in this PR was for resize operations. Not writing metadata is the exception and should be explicitly requested with `writeMetadata = false`, so in all other cases we write metadata.